enhancements/single-node-developer-cluster-profile.md

```diff
@@ -0,0 +1,514 @@
+  deployment. Single node clusters for non-developer use cases, such
+  as for edge computing, would require distinct cluster profiles with
+  independent selection and configuration of components specific to
+  those use cases.
+* A new CRC installer. Although it should somewhat simplify the SNC
+  tooling used for CRC, this enhancement proposal does not change the
+  basic approach to installing CRC or producing a VM image for the CRC
+  installer. By making the definition of the content and configuration
+  of the components in a single node developer cluster more orthogonal
+  to the tooling that deploys the cluster, this enhancement could
+  enable distinct future enhancements to improve the build and/or
+  install process used by CRC.
+* A new OpenShift installer. This enhancement proposal does not add a
+  new OpenShift installer or involve changes, beyond the ability to
+  specify the cluster profile, to the standard OpenShift installer
+  currently used to build CRC installation images. Evolution of the
+  existing installer, SNC tooling and other CRC components is
+  expected.
+* Functional changes for OpenShift. This enhancement proposal does not
+  result in any functional changes to clusters that do not utilize the
+  single-node-developer cluster profile.
+
+## Proposal
+
+The single-node-developer cluster profile will be implemented in two
+phases. The first phase introduces the new cluster profile, and
+enables some simplification of the SNC image build process by
+utilizing the profile to control which components are included in the
+cluster. The second phase focuses on optimizing the resource
+utilization of the components included in the cluster and may further
+simplify the SNC image build process.
+
+### Phase One
+
+In the initial phase, the following annotation will be added to the
+manifest files for all components that are to be included in single
+node developer clusters:
+
+```
+include.release.openshift.io/single-node-developer=true
+```
+
+This annotation will need to be added to the manifests for all the
+OpenShift components that are not currently removed by the snc.sh
+script during the CRC build process. The components that are currently
+removed, and thus should not need the annotation added, include:
+
+* most operators running in the openshift-monitoring namespace
+* most operators running in the openshift-insights namespace
+* machine-api-operator
+* Machine-config-operator (may need to run once, requiring the annotation)
+* openshift-cloud-credential-operator
+* openshift-cluster-storage-operator
+
+Additional components that are not currently removed by the SNC
+tooling might be also be determined to be unnecessary, and thus not
+require the annotation. Possibilities include:
+
+* openshift-cluster-node-tuning-operator
+* openshift-multus
+
+Once PRs adding this annotation have been merged to all the necessary
+component repositories, the snc.sh script will be updated to tell the
```

Comment:
Can we be more precise here and say: `the snc.sh script will be updated to tell the OpenShift installer to use the single-node-developer cluster profile via a new field ClusterProfile in the ClusterVersion object. This new field won't be used if an other profile is passed to the CVO by an env. variable.`

Reply:
> While looking at [openshift/cluster-version-operator#404](https://github.com/openshift/cluster-version-operator/pull/404) where we implement cluster profile as an environment variable mostly for IBM Cloud, we are missing for CRC a way to use this profile through the installer.
> 
> I suggest that we add a section saying that in the `ClusterVersion` object will contain an new immutable field in his spec: the profile.
> The CVO will load by order of precedence:
> 
> * the environment variable
> * the profile from the ClusterVersion.

This enhancement definitely requires some mechanism to specify the cluster profile to the installer and have that determine which cluster profile the CVO uses. I was hoping for some feedback on whether that mechanism should be addressed as part of this enhancement, or considered part of https://github.com/openshift/enhancements/blob/master/enhancements/update/cluster-profiles.md that this enhancement depends on. But that enhancement states "NOTE: The mechanism by which the environment variable is set on the CVO deployment is out of the scope of this design", so, unless someone identifies an existing enhancement that covers this, I'll include it in the next update to this PR.

Reply:
Given that https://github.com/openshift/enhancements/pull/504 defines a different cluster profile that also depends on being able to specify the profile to the installer, it probably makes most sense to cover adding that functionality in a separate enhancement PR.